draft-pauly-dprive-oblivious-doh.md

```diff
@@ -164,20 +170,28 @@ Clients MUST set the HTTP Content-Type header to "application/oblivious-dns-mess
 to indicate that this request is an Oblivious DoH query intended for proxying. Clients
 also SHOULD set this same value for the HTTP Accept header.
 
-Proxies MUST check that Client requests are correctly encoded, and MUST return a
-4xx (Client Error) if the check fails, along with the Proxy-Status response header
-with an "error" parameter of type "http_request_error" {{!I-D.ietf-httpbis-proxy-status}}.
 A correctly encoded request has the HTTP Content-Type header "application/oblivious-dns-message",
 uses the HTTP POST method, and contains "targethost" and "targetpath" variables.
+The Proxy MUST validate these parameters and construct a URI Template as follows:
 
-The "targethost" and "targetpath" variables are used to construct the request to forward to
-the Target. The Proxy is expected to send the Client's request using the URI
-constructed as "https://targethost/targetpath".
+1. Let $TARGET be the "targethost" parameter from the Client's request, encoded as the
+concatenation of a "host" value (Section 3.2.2 of {{!RFC3986}}) and, optionally, the
+":" character followed by port (Section 3.2.3 of {{!RFC3986}}). If this parameter cannot
+be encoded as such, the Client's request is incorrectly encoded.
+1. Let $PATH be the "targetpath" parameter from the Client's request, which MUST be a URI
+Template {{!RFC6570}}. If this parameter is not a valid URI Template, the Client's request
+is incorrectly encoded.
+1. The Target request HTTPS URI Template is "https://$TARGET$PATH".
```

Comment:
I don't understand why a URI is constructed here, just say that the proxy constructs a new request with the host and path it extracted - if you construct a URI then parse again you open yourself up to confusion

Reply:
Fixed.